src/components/dashboard/ConsentSteps.tsx

```diff
@@ -41,27 +43,35 @@ const ConsentSteps: React.FunctionComponent<ConsentStepsProps> = ({
     }
   }
 
+  const setLocalState = (clientData: { checkpoint: number }) => {
+    checkRedirectToDownload(clientData as any)
+    setUserClientData(clientData)
+    const { checkpoint } = clientData
+    if (checkpoint > 1) {
+      setStep(checkpoint)
+    }
+  }
+
   useEffect(() => {
     const getInfo = async () => {
```

Comment:
this is a start, i think we could remove this getInfo call entirely. 
but I'll let you take the decision @rfrabasile  :)